draft-private-access-tokens.md

```diff
@@ -152,59 +145,125 @@ origins could then track the client's federator user ID or the client's IP
+|             (validate)                     |          |
+|             (attach state)                 |          |
+|                    TokenRequest --->       |          |    PAT
+|                                 (validate) |   PAT    | Challenge/
+|                                 (evaluate) | Issuance |  Response
+|                    <--- TokenResponse      |   Flow   |   Flow
+|             (evaluate)                     |          |
+|             (update state)                 |          |
+|   <--- TokenResponse                       |          |
+---------------------------------------------/          |
+                                                        |
+     Response -------------------------------------- >  /
+~~~
+{: #fig-overview title=" PAT Architectural Components"}
+
+
+## Requirements
+
+In this architecture, the Mediator, Issuer, and Origin each have partial
+knowledge of the Client's identity and actions, and each entity only knows
+enough to serve its function (see {{terms}} for more about the pieces of
+information):
+
+- The Mediator knows the Client's identity (CLIENT_ID), the Issuer being
+  targeted (ISSUER_NAME), and the period of time for which the Issuer's policy
+  is valid (ISSUER_POLICY_WINDOW). The Mediator does not know the identity of
+  the Origin the Client is trying to access (ORIGIN_ID), but knows a
+  Client-anonymized identifier for it (ANON_ORIGIN_ID).
+
+- The Issuer knows the Origin's policy about client access and sees the Origin's
+  identity (ORIGIN_NAME) and the number of previous tokens issued to the Client
+  (as communicated by the Mediator), but the Client's identity is not visible to
+  the Origin.
+
+- The Origin knows the Issuer to delegate an incoming Client to (ISSUER_NAME),
+  and can verify that any tokens presented by the Client were signed by the
+  Issuer.
+
+Since an Issuer enforces policies on behalf of Origins, a Client is required to
+reveal the Origin's identity to the delegated Issuer. It is a requirement of
+this architecture that the Mediator not learn the Origin's identity so that,
+despite knowing the Client's identity, a Mediator cannot track and concentrate
+information about Client activity.
+
+An Issuer expects a Mediator to verify its Clients' identities correctly, but an
+Issuer cannot confirm a Mediator's efficacy or the Mediator-Client relationship
+directly without learning the Client's identity. Similarly, an Origin does not
+know the Mediator's identity, but ultimately relies on the Mediator to correctly
+verify or authenticate a Client for the Origin's policies to be correctly
+enforced. An Issuer therefore chooses to issue tokens to only known and
+reputable Mediators; the Issuer can employ its own methods to determine the
+reputation of a Mediator.
+
+A Mediator is expected to employ a stable Client identifier that can serve as a
+reasonable proxy for a user with some creation and maintenance cost on the user.
```

Comment:
We should probably also add something about client state here, which is required to be kept for origins (ANON_ORIGIN_ID) and the mediator (CLIENT_SECRET).

Reply:
Good point, will add something.

Reply:
@chris-wood -- added paragraph below, ptal.

Reply:
Looks great!